qcodes/instrument/meta/meta_parameter.py

```diff
@@ -0,0 +1,174 @@
+from qcodes.instrument.group_parameter import Group
+from typing import Any, Dict, Optional, Tuple, Union
+from collections import namedtuple
+from qcodes.instrument.parameter import (
+    ParamDataType,
+    ParamRawDataType,
+    Parameter,
+    _BaseParameter
+)
+
+import logging
+
+_log = logging.getLogger(__name__)
+
+
+class MetaGroup(Group):
+    """Group of Meta parameters"""
+    def __init__(
+        self,
+        name: str,
+        parameters: Tuple['MetaParameter'],
+        parameter_names: Tuple[str] = None,
+        **kwargs
+    ):
+        super().__init__(parameters=list(parameters), single_instrument=False, **kwargs)
+        parameter_names = parameter_names or [_e.name for _e in parameters]
+        self._namedtuple = namedtuple(name, parameter_names)
+
+    def get_parameters(self) -> Any:
+        return self._namedtuple(*[_p.get() for _p in self.parameters.values()])
+
+    def _set_from_dict(self, calling_dict: Dict[str, ParamRawDataType]) -> None:
+        for name, p in list(self.parameters.items()):
+            p.set(calling_dict[name])
+
+
+class MetaParameter(_BaseParameter):
+    """
+    Meta parameter that returns one or more endpoint parameter values
+    
+    Args:
+        name: Meta parameter name
+        endpoints: One or more endpoint parameters to alias
+        endpoint_names: One or more endpoint names
+        unit: The unit of measure. Use ``''`` for unitless.
+        setter: Optional setter function to override
+            endpoint parameter setter. Defaults to None.
+    """
+    def __init__(
+        self,
+        name: str,
+        endpoints: Tuple[Parameter],
+        endpoint_names: Tuple[str],
+        unit: str = None,
+        setter: callable = None,
+        **kwargs
+    ):
+        super().__init__(name, **kwargs)
+        self._endpoints = endpoints
+        self._setter = setter
+        self.unit = unit if unit is not None else ''
+        endpoint_names = endpoint_names or [_e.name for _e in endpoints]
+        self._group = None
+
+        # Generate sub parameters if >1 endpoints,
+        # e.g. my_meta_param.X, my_meta_param.Y
+        if len(endpoints) > 1:
+            self._parameters = self._sub_parameters(endpoints, endpoint_names)
+        self._set = self.set
+        self.set = self._set_override
```

Comment:
Yeah if I could just override `.set()` I would have but unfortunately that's not possible because of how this is implemented:
https://github.com/QCoDeS/Qcodes/blob/master/qcodes/instrument/parameter.py#L329

I suggest we just change that, and implement `set` as a method on `Parameter`. I a not sure if that would break anything else though.